pkg/kubectl/cmd/swagger_doc.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"io"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/kubernetes/pkg/kubectl"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+// NewCmdSwaggerDoc returns a cobra command for swagger docs
+func NewCmdSwaggerDoc(f *cmdutil.Factory, out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "doc MODEL-NAME...",
+		Short: "Print model's documentation through swagger's API.",
+		Run: func(cmd *cobra.Command, args []string) {
+			err := RunSwaggerDoc(f, out, cmd, args)
+			cmdutil.CheckErr(err)
+		},
+	}
+	cmd.Flags().BoolP("search", "S", false, "Search for similar models to given MODEL-NAMEs")
+	cmd.Flags().String("api-version", "", "The Kubernetes' API version we want to query")
+	return cmd
+}
+
+// RunSwaggerDoc executes the appropriate steps to print a model's documentation
+func RunSwaggerDoc(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string) error {
+	if len(args) == 0 {
+		return cmdutil.UsageError(cmd, "a MODEL-NAME is required to find documentation")
+	}
+
+	client, err := f.Client()
+	if err != nil {
+		return err
+	}
+
+	apiV := cmdutil.GetFlagString(cmd, "api-version")
+
+	for _, model := range args {
+		if cmdutil.GetFlagBool(cmd, "search") {
+			kubectl.SearchForModel(model, apiV, out, client)
+		} else {
```

Comment:
Make `GetModelDescription` and `SearchForModel` return errors instead of `os.Exit` so they can be used directly in return statements (will make this else clause redundant).